Plots/Polygons/NCL_polyg_2.py

```diff
@@ -0,0 +1,132 @@
+"""
+NCL_polyg_2.py
+==============
+Concepts illustrated:
+  - Drawing a Lambert Conformal U.S. map color-coded by climate divisions
+  - Color-coding climate divisions based on precipitation values
+  - Drawing the climate divisions of the U.S.
+  - Zooming in on a particular area on a Lambert Conformal map
+  - Drawing a border around filled polygons
+  - Masking the ocean in a map plot
+  - Masking land in a map plot
+  - Increasing the font size of text
+  - Adding text to a plot
+  - Drawing a custom labelbar on a map
+  - Creating a red-yellow-blue color map
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/polyg_2.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/polyg_2_lg.png
+"""
+
+###############################################################################
+# Import packages:
+# ----------------
+import xarray as xr
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import matplotlib.cm as cm
+from geocat.viz import util as gvutil
+
+import geocat.datafiles as gdf
+import cartopy.io.shapereader as shpreader
+import shapely.geometry as sgeom
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+
+###############################################################################
+# Initialize color map and bounds for each color
+
+colormap = colors.ListedColormap(['mediumpurple', 'mediumblue', 'royalblue', 'cornflowerblue', 'lightblue', 'teal', 'yellowgreen',
+                                  'green', 'wheat', 'tan', 'gold', 'orange', 'red', 'firebrick'])
+
+colorbounds = [0, 5, 10, 15, 20, 25, 30, 35, 40, 50, 60, 70, 80, 90, 100]
+
+###############################################################################
+# Define helper function to determine which color to fill the divisions based on precipitation data
+
+
+def findDivColor(colorbounds, pdata):
```

Comment:
What gets returned by this function if the 'else' clause is never executed (if pdata is outside of range of colorbounds)?

Reply:
@clyne I added a case where if it's bigger than the last color bound listed, it still returns the last color in the colorbar. Is this ok?

Reply:
> Just need to try to figure out why the two regions in the north west (washington and oregon) are getting mapped to different colors than the NCL version.

I am still confused about the colors of the original NCL plot... I thought I was making a rounding error, but now I'm not sure that is the case. 

In the original plot the 4th climate division in Oregon (the uppermost leftmost division) is pink even though the precipitation data is 93.402, which falls within the 90-100 range (which is supposed to be dark red). 

Additionally, the rounding for the NCL bounds doesn't seem to be consistent. For example, for the 10th division in Texas, the precipitation value is 25.3749 and the color shows it is in the 25-30 range. However, for the 4th division in Oregon, the precipitation value is 70.389 and the color shows it is in the 60-70 range. Not sure how they decided which divisions to round up/round down.

Reply:
Are you sure your script is calculating the precipitation data the same way the NCL script is?  They look very different to me.

Reply:
> Are you sure your script is calculating the precipitation data the same way the NCL script is? They look very different to me.

Yes I was! I was getting the average annual precipitation by dividing the data value by 100 instead of 101. I think it's fixed now. 

Here's how the plot looks now:

![polyg_2 graph](https://user-images.githubusercontent.com/36773444/83705481-e9ef3900-a5c9-11ea-8d81-774fb8de7902.PNG)

Reply:
Nice!